app/models/book.rb

```diff
@@ -0,0 +1,61 @@
+class Book < ApplicationRecord
+  CATEGORIES = %w[mobile_development photo web_design web_development].freeze
+
+  has_and_belongs_to_many :authors
+  has_and_belongs_to_many :categories
+  has_many :images, dependent: :destroy
+  has_many :reviews, dependent: :destroy
+  has_many :line_items
+
+  before_destroy :ensure_not_referenced_by_any_line_item
+
+  validates :title, :description, :price, :publication_year,
+            :dimensions, :materials, presence: true
+  validates :price, numericality: { greater_than_or_equal_to: 0.01 }
+  validates :title, uniqueness: { case_sensitive: false }
+  validates :publication_year, inclusion: { in: 1969..Date.today.year }
+
+  paginates_per 12
+
+  scope :mobile_development, -> do
```

Comment:
@b-artem these scopes also look strange, what should we do if we need to add one more category about 'DB architecture' for example

Reply:
Deleted these scopes. `Category.find_by(name: category_name).books` could do the same